tests/unit/test_http_clients.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+from contextlib import asynccontextmanager
+
+import pytest
+from aiodynamo.client import Client
+from aiodynamo.credentials import Key, StaticCredentials
+from aiodynamo.http.aiohttp import AIOHTTP
+from aiodynamo.http.base import RequestFailed
+from aiodynamo.models import ExponentialBackoffThrottling
+from yarl import URL
+
+pytestmark = [pytest.mark.asyncio]
+
+
+async def test_aiohttp_request_failed():
+    exc = asyncio.TimeoutError()
+
+    class FakeSession:
+        @asynccontextmanager
+        async def request(self, *args, **kwargs):
+            raise exc
+            yield
+
+    http = AIOHTTP(FakeSession())
+
+    with pytest.raises(RequestFailed) as error:
+        await http.get(url=URL("http://foo.invalid.com"), timeout=100)
+    assert error.value.__cause__ is exc
```

Comment:
I wonder if it's enough to support both `__cause__` and `__context__` as in PEP-3134:
https://www.python.org/dev/peps/pep-3134/#enhanced-reporting

Reply:
From how I  understand it, `__cause__` is for explicit chaining,  `__context__` for implicit. Since this PR changes to explicit chaining, checking for `__cause__` seems reasonable.

Reply:
I guess what puzzles me is why does `aiodynamo` care?
The library evaluates neither `__cause__` nor `__context__`, so what's the rationale for the requirement?
In other words, why can't the transport use implicit (default) chaining?